tripy/tripy/frontend/trace/ops/resize.py

```diff
@@ -0,0 +1,152 @@
+                rank=1,
+                dtype=float32,
+                device=outputs[0].device,
+                reason_details=[f"create scales tensor in resize op."],
+            )
+            ConstantOp.build([], [scales_tensor], data=self.scales)
+            # inputs[1] is input[0].shape
+            input_shape_f32 = FlatIRTensor.build(
+                shape=out_shape,
+                rank=1,
+                dtype=float32,
+                device=outputs[0].device,
+                reason_details=[f"convert input shape tensor to float32 in resize op."],
+            )
+            ConvertOp.build([inputs[1]], [input_shape_f32])
+            out_shape_f32 = FlatIRTensor.build(
+                shape=out_shape,
+                rank=1,
+                dtype=float32,
+                device=outputs[0].device,
+                reason_details=[f"compute output shape in resize op."],
+            )
+            MulOp.build([input_shape_f32, scales_tensor], [out_shape_f32])
+            out_shape_tensor = FlatIRTensor.build(
+                shape=out_shape,
+                rank=1,
+                dtype=int32,
+                device=outputs[0].device,
+                reason_details=[f"convert output shape to int32 in resize op."],
+            )
+            ConvertOp.build([out_shape_f32], [out_shape_tensor])
+            inputs[1] = out_shape_tensor
+
+        if self.mode == "nearest":
+            ResizeNearestOp.build(inputs, outputs)
+        elif self.mode == "cubic":
+            ResizeCubicOp.build(inputs, outputs, self.align_corners, -0.75)
+        else:
+            ResizeLinearOp.build(inputs, outputs, self.align_corners)
+
+
+@export.public_api(document_under="operations/functions")
+@constraints.dtype_info(
+    dtype_variables={
+        "T1": ["float32", "float16", "int8"],
+    },
+    dtype_constraints={"input": "T1", constraints.RETURN_VALUE: "T1"},
+)
+@frontend_utils.convert_shape_inputs(["output_shape"])
+def resize(
+    input: "tripy.Tensor",
+    mode: str,
+    output_shape: "tripy.types.ShapeLike" = None,
+    scales: Sequence[Union[int, float]] = None,
+    align_corners: bool = False,
+) -> "tripy.Tensor":
+    r"""
+    Resizes the input tensor.
+
+    Args:
+        input: The input tensor.
+        mode: The resize operation's algorithm. Must be one of "cubic, linear, nearest".
+        output_shape: The output shape of the resize operation.
+        scales: A sequence of scale factors for each dimension. Must have
+            the same length as input tensor's rank. Will be ignored if ``output_shape`` is
+            given.
+        align_corners: If set to ``True``, the input and output tensors are aligned by the
+            center points of their corner pixels, preserving the values at the corner pixels.
+            If set to ``False``, the input and output tensors are aligned by the corner points of
+            their corner pixels. Only in effect when ``mode`` is ``cubic`` or ``linear``.
```

Comment:
Could we emit an error or a warning when this is set with an unsupported mode?

Reply:
Added, it will raise an error, same as `torch`.